[DO_NOT_MERGE] Fix PipeLinePMDAO

hchoura opened the pull request.

hchoura committed:
Fix PipeLinePMDAO

hchoura committed:
Merge branch 'development' into FixPipelinePMDAO

hchoura committed:
add nullPM support

jlhughes commented on src/foam/nanos/pm/NullPM.js:
```diff
@@ -29,7 +29,16 @@ foam.CLASS({
         }
       ],
-      javaCode: '/*nop*/'
+      javaCode: `
```
@hchoura what's reason for an implementation?

hchoura replied:
To activate the log.

jlhughes replied:
Please explain.  If you need PMs for every create, then change the percentage in the nspec, it's presently set to 1%, change it 100. 

foam/nanos/pm/services.jrl

p({
  class:"foam.nanos.boot.NSpec",
  name:"PM",description:"Configurable PM Factory",
  authenticate:false,
  pm:false,
  service:{
    class:"foam.nanos.pm.PMFactory",
    percentage:1
  }
})

hchoura committed:
add synchronized

formatting

jlhughes commented on src/foam/nanos/pm/PMInfo.js:
```diff
@@ -80,6 +80,7 @@ foam.CLASS({
 
   methods: [
     {
+      synchronized: true,
```
What issue are you seeing? synchronization in PMs is the last thing we need as they need to be super fast.

hchoura committed:
Merge branch 'development' into FixPipelinePMDAO

jlhughes committed:
On Consensus delay, only request Replay once per active Alarm.
To for replay, clear alarm.

jlhughes committed:
Merge pull request #751 from kgrgreer/NP-5201/MedusaConsenusReplayRequestCmd

On Consensus delay, only request Replay once per active Alarm.

jlhughes committed:
Add singleton support to StdoutLogger

Also, have it setup the same logging DAO stack (sans JDAO) as normal
logging, so StdoutLogger and StdoutLoggerDAO have the same output/layout.

kgrgreer committed:
Merge pull request #752 from kgrgreer/NP-5242/StdoutLogger

Add singleton support to StdoutLogger

hchoura committed:
fix TextSearchView

kgrgreer committed:
Merge pull request #755 from kgrgreer/FixTextSearchView

fix TextSearchView

hchoura committed:
Merge branch 'development' into FixPipelinePMDAO
